Implement DeliveryDelay, Priority and TimeToLive in PheanstalkProducer

likeuntomurphy opened the pull request:
I could not take advantage of the `MultiplierRetryStrategy` built in to the Symfony Messenger component, because the Pheanstalk producer did not fully implement the `setDeliveryDelay` method (see also #821). Even though I only needed the delay, I went ahead and fully implemented the priority and time to live functionality of the `Producer` interface as well.

The code changes in the PR essentially do three things:

1.  ensures that “delivery delay” and “time to live” are set as milliseconds on the producer but passed to the transport as seconds; 
2. ensures that the respective message values are used when priority, delay or TTL is null; and
3. ensures that any producer values do not persist for more than one send.

Each of these points has test coverage.

makasim commented:
Thank you for the contribution @likeuntomurphy. 

> ensures that any producer values do not persist for more than one send.

That's not true. In fact, the producer should use the values for all messages that go through it. That's home JMS defines it and other Enqueue transports work.

likeuntomurphy commented:
Thanks for the feedback, @makasim.

I realized that the enqueue transport will set the values on the producer for each send as long as the values are configured as options (https://github.com/sroze/messenger-enqueue-transport/blob/master/QueueInteropTransport.php#L140). So I removed the code that was resetting the values after each send.

And I was also wrong about how to prioritize the values. I copied a block of code from the Mongo producer and modified it for Beanstalk, and I updated the tests accordingly.
